### PR TITLE
WIP - latest coach tables work

### PR DESCRIPTION
**Pull request by indirectlylit**

### CODE REVIEW

File: kolibri/core/assets/src/vue/content-icon/content-icons/user.svg [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<svg width="14px" height="14px" viewBox="0 0 14 14" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
```
Comment (christianmemije) [inline kolibri/core/assets/src/vue/content-icon/content-icons/user.svg]: Why not use this icon instead? https://material.io/icons/#ic_person

File: kolibri/core/assets/src/vue/content-icon/index.vue [hunk 2]
```diff
@@ -23,14 +24,15 @@
...
+      user: 'user',
     },
     props: {
       kind: {
         type: String,
         required: true,
         validator(value) {
           for (const contentKind in ContentKinds) {
-            if (value === ContentKinds[contentKind] || 'topic') {
+            if (value === ContentKinds[contentKind] || 'topic' || 'user') {
```
Comment (christianmemije) [inline kolibri/core/assets/src/vue/content-icon/index.vue]: Reference constants instead?

File: kolibri/core/assets/src/vue/content-icon/content-icons/user.svg (hunk 1 above)
Comment (indirectlylit) [reply kolibri/core/assets/src/vue/content-icon/content-icons/user.svg]: good catch. I'd copied the icon from another place in the app - i'll update them both

File: kolibri/plugins/coach_tools/assets/src/actions.js [hunk 3]
```diff
@@ -16,35 +17,18 @@ const Constants = require('./state/constants');
...
-function _useReportPageLoadingFlag(newParams, oldParams) {
-  if (!newParams || !oldParams) {
-    return true;
-  }
-  if (Object.entries(newParams).length !== Object.entries(newParams).length) {
-    return true;
+/* find the keys that differ between the old and new params */
+function _diffKeys(newParams, oldParams) {
+  if (!oldParams) {
+    return Object.entries(newParams).map(([key, value]) => key);
```
Comment (christianmemije) [inline kolibri/plugins/coach_tools/assets/src/actions.js]: Make use of the function you created `enumerate()`

File: kolibri/plugins/coach_tools/assets/src/actions.js [hunk 4]
```diff
@@ -109,34 +93,33 @@ function showReport(store, params, oldParams) {
...
-  store.dispatch('SET_CONTENT_SCOPE_ID', contentScopeId);
-  store.dispatch('SET_USER_SCOPE', userScope);
-  store.dispatch('SET_USER_SCOPE_ID', userScopeId);
-  store.dispatch('SET_ALL_OR_RECENT', allOrRecent);
-  store.dispatch('SET_VIEW_BY_CONTENT_OR_LEARNERS', viewByContentOrLearners);
-  store.dispatch('SET_SORT_COLUMN', sortColumn);
-  store.dispatch('SET_SORT_ORDER', sortOrder);
-
+  // set loading true for all except 'view-by', which just affects table (not summary)
+  if (diffKeys.length === 1 && diffKeys[0] !== 'view_by_content_or_learners') {
```
Comment (christianmemije) [inline kolibri/plugins/coach_tools/assets/src/actions.js:121]: Should instead be `if (diffKeys.length && diffKeys[0] !== 'view_by_content_or_learners')`

File: kolibri/plugins/coach_tools/assets/src/state/getters.js [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    );
+  },
+  dataTable(state) {
+    let data = [];
+
+    // CONTENT or LEARNERS
+    if (state.pageState.view_by_content_or_learners === Constants.ViewBy.CONTENT) {
+      data = state.pageState.table_data.map(item => ({
+        kind: item.kind,
+        id: item.pk,
```
Comment (christianmemije) [inline kolibri/plugins/coach_tools/assets/src/state/getters.js:80]: Why no `toString()` here, as you have below.

File: kolibri/plugins/coach_tools/assets/src/actions.js (hunk 3 above)
Comment (indirectlylit) [reply kolibri/plugins/coach_tools/assets/src/actions.js]: enumerate actually returns values, whereas this one returns keys.

In both cases, I think I'll switch to lodash.

File: kolibri/plugins/coach_tools/assets/src/actions.js (hunk 4 above)
Comment (indirectlylit) [reply kolibri/plugins/coach_tools/assets/src/actions.js:121]: The `=== 1` is actually necessary to ensure that _only_ that key changed. I'll update the comment to clarify.

File: kolibri/plugins/coach_tools/assets/src/state/getters.js (hunk 5 above)
Comment (indirectlylit) [reply kolibri/plugins/coach_tools/assets/src/state/getters.js:80]: This one is already a string - the other one is an integer.

Filing a related issue here: #657 - if we can standardize on strings, we can take out the `toString` logic.

File: kolibri/plugins/coach_tools/assets/src/actions.js (hunk 3 above)
Comment (christianmemije) [reply kolibri/plugins/coach_tools/assets/src/actions.js]: My mistake. :+1:

File: kolibri/core/assets/src/vue/content-icon/content-icons/user.svg (hunk 1 above)
Comment (indirectlylit) [reply kolibri/core/assets/src/vue/content-icon/content-icons/user.svg]: only updating this copy. Filed a new issue for the other here: #658

File: kolibri/plugins/coach_tools/assets/src/actions.js (hunk 3 above)
Comment (indirectlylit) [reply kolibri/plugins/coach_tools/assets/src/actions.js]: apparently there's `.keys` method I wasn't aware of! https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys

**Pull request merged**